### ISSUE TITLE
[q] inline favicon in vite

### ISSUE THREAD

**Issue by ZuBB** [posthtml/posthtml-inline-favicon#44]
Hey there!

I need to inline the favicon (PNG image) into a `.html` file in a project that uses Vite as a bundler.

Right now, I have the next snippet in the config:

```js
...
  const plugins = [
    react(), svgr(), Info(),
    posthtml({
      plugins: [
        inlineFavicon()
      ]
    })
  ];
...
```

And it does not work.

Here is a source line from my HTML file:

```html
...
    <link rel="icon" type="image/png" sizes="32x32" href="/src/assets/favicon.png">
...
```

This is what I get in resulted HTML file:
```html
...
    <link rel="icon" type="image/png" sizes="32x32" href="./favicon-Co0w4LGE.png">
...
```

Any chance you have some ideas/hints on this issue?

**Comment by Scrum** [comment]
Hi, can you create the simplest example of reproducing this? It is not very clear what kind of configuration you have

**Comment by ZuBB** [comment]
Ok, I figured out why it did not work for me. It's because the selector needs a `shortcut` value.

https://github.com/posthtml/posthtml-inline-favicon/blob/8658d2c2553c7ed45ea41fb7aa8e9762d9c585d7/src/inlineFavicon.ts#L8

Accordingly, to mdn, `shortcut` is an old link type and should not be used.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel#sect4

Are you interested in this change? It should not be very hard. I can try to cook a PR with it

**Comment by metonym** [comment]
@ZuBB A PR would be most welcome, thank you! Maybe `rel` could just be a RegExp for "icon"

**Comment by ZuBB** [comment]
> Maybe `rel` could just be a RegExp for "icon"

Yes, that was my plan.

Can we expect a new release after the fix will be applied? If so, can you give an ETA for that?

**Comment by metonym** [comment]
As soon as it's applied, I'm available to re-publish. Will be around for an hour or so; otherwise, will do so first thing in the morning.

**Comment by ZuBB** [comment]
> As soon as it's applied, I'm available to re-publish. Will be around for an hour or so; otherwise, will do so first thing in the morning.

Great. already working on the change(s)

**Comment by ZuBB** [comment]
here it is @metonym 

https://github.com/posthtml/posthtml-inline-favicon/pull/45

**Comment by ZuBB** [comment]
Thanks a lot, @metonym, for the quick help.

**Comment by metonym** [comment]
Released in [v0.1.3](https://github.com/posthtml/posthtml-inline-favicon/releases/tag/v0.1.3). Thanks for the fix!
